patches/ungoogled-chromium/macos/fix-gn-safe_browsing.patch

```diff
@@ -2,42 +2,74 @@
 
 --- a/build/config/compiler/BUILD.gn
 +++ b/build/config/compiler/BUILD.gn
-@@ -1478,7 +1478,7 @@ config("default_warnings") {
+@@ -1512,7 +1512,7 @@ config("default_warnings") {
          # recognize.
          cflags += [
            # TODO(thakis): https://crbug.com/753973
 -          "-Wno-enum-compare-switch",
 +          # "-Wno-enum-compare-switch",
-         ]
-       }
-     }
+
+           # Ignore warnings about MSVC optimization pragmas.
+           # TODO(thakis): Only for no_chromium_code? http://crbug.com/505314
 --- a/chrome/browser/BUILD.gn
 +++ b/chrome/browser/BUILD.gn
-@@ -1674,7 +1674,6 @@ jumbo_split_static_library("browser") {
-     "//chrome/browser/net:probe_message_proto",
+@@ -1704,7 +1704,6 @@ jumbo_split_static_library("browser") {
      "//chrome/browser/profiling_host",
+     "//chrome/browser/push_messaging:budget_proto",
      "//chrome/browser/resource_coordinator:mojo_bindings",
 -    "//chrome/browser/safe_browsing",
      "//chrome/browser/ssl:proto",
      "//chrome/browser/ui",
      "//chrome/browser/ui/webui/bluetooth_internals",
 --- a/chrome/browser/extensions/BUILD.gn
 +++ b/chrome/browser/extensions/BUILD.gn
-@@ -800,7 +800,6 @@ jumbo_static_library("extensions") {
+@@ -808,7 +808,6 @@ jumbo_static_library("extensions") {
      "//chrome/browser/media/router",
      "//chrome/browser/media/router/discovery",
      "//chrome/browser/resource_coordinator:mojo_bindings",
 -    "//chrome/browser/safe_browsing",
+     "//chrome/browser/web_applications/components",
      "//chrome/browser/web_applications/extensions",
      "//chrome/common",
-     "//chrome/common/extensions:mojo_bindings",
 --- a/chrome/browser/ui/BUILD.gn
 +++ b/chrome/browser/ui/BUILD.gn
-@@ -910,7 +910,6 @@ split_static_library("ui") {
+@@ -837,7 +837,6 @@ jumbo_split_static_library("ui") {
      "//chrome/browser/engagement:mojo_bindings",
      "//chrome/browser/media:mojo_bindings",
      "//chrome/browser/profiling_host",
 -    "//chrome/browser/safe_browsing",
      "//chrome/browser/ssl:proto",
      "//chrome/browser/ui/webui/bluetooth_internals",
      "//chrome/browser/ui/webui/interventions_internals:mojo_bindings",
+--- a/chrome/browser/ui/webui/settings/change_password_handler.cc
++++ b/chrome/browser/ui/webui/settings/change_password_handler.cc
+@@ -52,19 +52,19 @@
+ }
+ 
+ void ChangePasswordHandler::HandleChangePassword(const base::ListValue* args) {
```

Comment:
libui.a static framework refers below two functions from safe browsing class..
safe_browsing::ChromePasswordProtectionService::OnUserAction
safe_browsing::ChromePasswordProtectionService::ShouldShowChangePasswordSettingUI

As we remove safe browsing reference using fix-gn-safe_browsing.patch file, which causes linking error..
So i removed corresponding safe browse reference from change password handler UI class..
i guess we could replace "ShouldShowChangePasswordSettingUI" with "true" boolean value..

below is the error:

> [19859/20025] LINK ./v8_context_snapshot_generator
warning: no debug symbols in executable (-arch x86_64)
[19871/20025] SOLINK 'obj/chrome/chrom...ts/Resources/DWARF/Chromium Framework'
FAILED: obj/chrome/chrome_framework_shared_library/Chromium Framework obj/chrome/chrome_framework_shared_library/Chromium Framework.TOC Chromium Framework.dSYM Chromium Framework.dSYM/Contents/Info.plist Chromium Framework.dSYM/Contents/Resources/DWARF/Chromium Framework 
if [ ! -e "obj/chrome/chrome_framework_shared_library/Chromium Framework" -o ! -e "obj/chrome/chrome_framework_shared_library/Chromium Framework.TOC" ] || otool -l "obj/chrome/chrome_framework_shared_library/Chromium Framework" | grep -q LC_REEXPORT_DYLIB ; then TOOL_VERSION=1539717229 ../../build/toolchain/mac/linker_driver.py ../../third_party/llvm-build/Release+Asserts/bin/clang++ -shared  -Wcrl,dsym,. -Wl,-install_name,@executable_path/../Versions/70.0.3538.67/Chromium\ Framework.framework/Chromium\ Framework -compatibility_version 3538.0.67 -current_version 3538.0.67 -Wl,-order_file,../../chrome/app/framework.order -stdlib=libc++ -arch x86_64 -segprot PROTECTED_MEMORY rw r -isysroot ../../../../../../../../../Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -mmacosx-version-min=10.9.0 -Wl,-ObjC -Wcrl,strip,-x,-S -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/lib -o "obj/chrome/chrome_framework_shared_library/Chromium Framework" -Wl,-filelist,"obj/chrome/chrome_framework_shared_library/Chromium Framework.rsp" -framework Cocoa -framework Foundation -framework IOKit -framework Security -framework SystemConfiguration -framework Accelerate -framework AudioUnit -framework DiskArbitration -framework ImageCaptureCore -framework OpenGL -framework QuartzCore -framework SecurityInterface -lcups -framework ApplicationServices -framework AppKit -lbsm -framework CoreFoundation -framework CFNetwork -framework CoreServices -lresolv -framework IOBluetooth -framework Carbon -framework CoreVideo -framework CoreGraphics -framework CoreText -framework IOSurface -framework AudioToolbox -framework CoreAudio -framework Quartz -framework AVFoundation -framework CoreMedia -framework VideoToolbox -lsandbox -framework CoreMIDI -framework OpenDirectory -framework AddressBook -framework LocalAuthentication -framework ForceFeedback -framework GameController -framework CoreWLAN  && { otool -l "obj/chrome/chrome_framework_shared_library/Chromium Framework" | grep LC_ID_DYLIB -A 5; nm -gP "obj/chrome/chrome_framework_shared_library/Chromium Framework" | cut -f1-2 -d' ' | grep -v U$$; true; } > "obj/chrome/chrome_framework_shared_library/Chromium Framework.TOC"; else TOOL_VERSION=1539717229 ../../build/toolchain/mac/linker_driver.py ../../third_party/llvm-build/Release+Asserts/bin/clang++ -shared  -Wcrl,dsym,. -Wl,-install_name,@executable_path/../Versions/70.0.3538.67/Chromium\ Framework.framework/Chromium\ Framework -compatibility_version 3538.0.67 -current_version 3538.0.67 -Wl,-order_file,../../chrome/app/framework.order -stdlib=libc++ -arch x86_64 -segprot PROTECTED_MEMORY rw r -isysroot ../../../../../../../../../Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -mmacosx-version-min=10.9.0 -Wl,-ObjC -Wcrl,strip,-x,-S -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/lib -o "obj/chrome/chrome_framework_shared_library/Chromium Framework" -Wl,-filelist,"obj/chrome/chrome_framework_shared_library/Chromium Framework.rsp" -framework Cocoa -framework Foundation -framework IOKit -framework Security -framework SystemConfiguration -framework Accelerate -framework AudioUnit -framework DiskArbitration -framework ImageCaptureCore -framework OpenGL -framework QuartzCore -framework SecurityInterface -lcups -framework ApplicationServices -framework AppKit -lbsm -framework CoreFoundation -framework CFNetwork -framework CoreServices -lresolv -framework IOBluetooth -framework Carbon -framework CoreVideo -framework CoreGraphics -framework CoreText -framework IOSurface -framework AudioToolbox -framework CoreAudio -framework Quartz -framework AVFoundation -framework CoreMedia -framework VideoToolbox -lsandbox -framework CoreMIDI -framework OpenDirectory -framework AddressBook -framework LocalAuthentication -framework ForceFeedback -framework GameController -framework CoreWLAN  && { otool -l "obj/chrome/chrome_framework_shared_library/Chromium Framework" | grep LC_ID_DYLIB -A 5; nm -gP "obj/chrome/chrome_framework_shared_library/Chromium Framework" | cut -f1-2 -d' ' | grep -v U$$; true; } > "obj/chrome/chrome_framework_shared_library/Chromium Framework.tmp" && if ! cmp -s "obj/chrome/chrome_framework_shared_library/Chromium Framework.tmp" "obj/chrome/chrome_framework_shared_library/Chromium Framework.TOC"; then mv "obj/chrome/chrome_framework_shared_library/Chromium Framework.tmp" "obj/chrome/chrome_framework_shared_library/Chromium Framework.TOC" ; fi; fi
Undefined symbols for architecture x86_64:
  "safe_browsing::ChromePasswordProtectionService::OnUserAction(content::WebContents*, safe_browsing::LoginReputationClientRequest_PasswordReuseEvent_ReusedPasswordType, safe_browsing::WarningUIType, safe_browsing::WarningAction)", referenced from:
      settings::ChangePasswordHandler::HandleChangePassword(base::ListValue const*) in libui.a(ui_jumbo_7.o)
  "safe_browsing::ChromePasswordProtectionService::ShouldShowChangePasswordSettingUI(Profile*)", referenced from:
      settings::ChangePasswordHandler::UpdateChangePasswordCardVisibility() in libui.a(ui_jumbo_7.o)
ld: symbol(s) not found for architecture x86_64
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
Traceback (most recent call last):
  File "../../build/toolchain/mac/linker_driver.py", line 229, in <module>
    Main(sys.argv)
  File "../../build/toolchain/mac/linker_driver.py", line 79, in Main
    subprocess.check_call(compiler_driver_args)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../../third_party/llvm-build/Release+Asserts/bin/clang++', '-shared', '-Wl,-install_name,@executable_path/../Versions/70.0.3538.67/Chromium Framework.framework/Chromium Framework', '-compatibility_version', '3538.0.67', '-current_version', '3538.0.67', '-Wl,-order_file,../../chrome/app/framework.order', '-stdlib=libc++', '-arch', 'x86_64', '-segprot', 'PROTECTED_MEMORY', 'rw', 'r', '-isysroot', '../../../../../../../../../Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk', '-mmacosx-version-min=10.9.0', '-Wl,-ObjC', '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/lib', '-o', 'obj/chrome/chrome_framework_shared_library/Chromium Framework', '-Wl,-filelist,obj/chrome/chrome_framework_shared_library/Chromium Framework.rsp', '-framework', 'Cocoa', '-framework', 'Foundation', '-framework', 'IOKit', '-framework', 'Security', '-framework', 'SystemConfiguration', '-framework', 'Accelerate', '-framework', 'AudioUnit', '-framework', 'DiskArbitration', '-framework', 'ImageCaptureCore', '-framework', 'OpenGL', '-framework', 'QuartzCore', '-framework', 'SecurityInterface', '-lcups', '-framework', 'ApplicationServices', '-framework', 'AppKit', '-lbsm', '-framework', 'CoreFoundation', '-framework', 'CFNetwork', '-framework', 'CoreServices', '-lresolv', '-framework', 'IOBluetooth', '-framework', 'Carbon', '-framework', 'CoreVideo', '-framework', 'CoreGraphics', '-framework', 'CoreText', '-framework', 'IOSurface', '-framework', 'AudioToolbox', '-framework', 'CoreAudio', '-framework', 'Quartz', '-framework', 'AVFoundation', '-framework', 'CoreMedia', '-framework', 'VideoToolbox', '-lsandbox', '-framework', 'CoreMIDI', '-framework', 'OpenDirectory', '-framework', 'AddressBook', '-framework', 'LocalAuthentication', '-framework', 'ForceFeedback', '-framework', 'GameController', '-framework', 'CoreWLAN']' returned non-zero exit status 1
ninja: build stopped: subcommand failed.

Reply:
I see, I didn't realize `service_` was referring to Safe Browsing code. Thanks for explaining.

> i guess we could replace "ShouldShowChangePasswordSettingUI" with "true" boolean value..

We would need to set the whole boolean statement to `true` in this scenario.

Could you check what `change-password-visibility` does from the Web UI JavaScript (since `FireWebUIListener` triggers a JavaScript event)? I'm not able to inspect the code at the moment.

Reply:
here chromium looking at account sync password and deciding whether to prompt for password change or not based on safe browsing settings.. 
If we replace with true, when safe browsing warning enabled, chromium always asks for password change.. may be we need to keep "false" by default.. so it will not prompt for password change..

related files:
chrome/browser/safe_browsing/chrome_password_protection_service_browsertest.cc:327
chrome/browser/ui/webui/settings/change_password_handler.cc:63
chrome/browser/resources/settings/basic_page/basic_page.js:127
chrome/browser/resources/settings/change_password_page/change_password_page.html
chrome/browser/resources/settings/basic_page/basic_page.html:134

Reply:
one more thing is as we commented onUserAction function contents using patch file, handling password change is no more available. so, it does not matter what value returns from "ShouldShowChangePasswordSetting"..

Reply:
I see. In that case, what you have in the patch is good. Thanks.